packages/sanity/src/core/studio/components/navbar/search/components/filters/filter/inputs/date/datePicker/calendar/CalendarDay.tsx

```diff
@@ -1,51 +1,118 @@
-import {Card, Text} from '@sanity/ui'
+import {Card, Text, Theme} from '@sanity/ui'
+import {isAfter, isBefore, isSameDay, isSameMonth} from 'date-fns'
 import React, {useCallback} from 'react'
-import {useDatePicker} from '../contexts/useDatePicker'
+import styled, {css} from 'styled-components'
+import {useCalendar} from './contexts/useDatePicker'
 
 interface CalendarDayProps {
   date: Date
-  focused?: boolean
   onSelect: (date: Date) => void
-  isCurrentMonth?: boolean
-  isToday: boolean
-  selected?: boolean
 }
 
-export function CalendarDay(props: CalendarDayProps) {
-  const {date, focused, isCurrentMonth, isToday, onSelect, selected} = props
+const CircleSvg = styled.svg(({theme}: {theme: Theme}) => {
+  return css`
+    bottom: 0;
+    left: 0;
+    position: absolute;
+    right: 0;
+    top: 0;
 
+    circle {
+      stroke: ${theme.sanity.color.card.enabled.border};
+      stroke-width: 3;
+      fill: none;
+    }
+  `
+})
+
+const CustomCard = styled(Card)<{
```

Comment:
I have a suggestion to simplify this a bit with data attributes, feel free to ignore if you don't agree :) 

I suggest adding `radius` to the `CustomCard` component in the JSX:

```tsx
<CustomCard
  // ... all the other props
  data-within-range={isWithinRange ? true : undefined}
  data-start-date={isStartDate ? true : undefined}
  data-end-date={isEndDate ? true : undefined}
  radius={2}
/>
```

And then rewrite the `CustomCard` styling to remove the radius, like this:

```ts
const CustomCard = styled(Card)`
  position: relative;

  &[data-focused='true'] {
    z-index: 1;
  }

  &[data-start-date='true'] {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }

  &[data-end-date='true'] {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }

  &[data-within-range='true'] {
    border-radius: 0;
  }
`
```